### PR TITLE
KAFKA-17609: Migrate broker compatibility test from ZK to KRaft

### PR DESCRIPTION
**Pull request by bbejeck**
*More detailed description of your change,
if necessary. The PR title and PR message become
the squashed commit message, so use a separate
comment to ping reviewers.*

*Summary of testing strategy (including rationale)
for the feature or bug fix. Unit and/or integration
tests are expected for any behaviour change and
system tests should be considered for larger changes.*

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: tests/kafkatest/tests/streams/streams_broker_compatibility_test.py [hunk 1]
```diff
@@ -18,34 +18,27 @@
...
+from kafkatest.version import LATEST_2_8, \
     LATEST_3_0, LATEST_3_1, LATEST_3_2, LATEST_3_3, LATEST_3_4, LATEST_3_5, LATEST_3_6, LATEST_3_7, LATEST_3_8, KafkaVersion
 
 
 class StreamsBrokerCompatibility(Test):
     """
     These tests validates that
-    - Streams works for older brokers 0.11 (or newer)
-    - Streams w/ EOS-alpha works for older brokers 0.11 (or newer)
     - Streams w/ EOS-v2 works for older brokers 2.5 (or newer)
```
Comment (mjsax) [inline tests/kafkatest/tests/streams/streams_broker_compatibility_test.py:30]:
```suggestion
    - Streams w/ ALOS and EOS-v2 works for older brokers 2.8 (or newer)
```

**Review by mjsax — APPROVED** [review]
One minor comment. Overall LGTM.

File: tests/kafkatest/tests/streams/streams_broker_compatibility_test.py [hunk 2]
```diff
@@ -60,17 +53,15 @@ def __init__(self, test_context):
...
-
 
     @cluster(num_nodes=4)
-    @matrix(broker_version=[str(LATEST_1_0),str(LATEST_1_1),str(LATEST_2_0),str(LATEST_2_1),
-                            str(LATEST_2_2),str(LATEST_2_3),str(LATEST_2_4),str(LATEST_2_5),
-                            str(LATEST_2_6),str(LATEST_2_7),str(LATEST_2_8),str(LATEST_3_0),
-                            str(LATEST_3_1),str(LATEST_3_2),str(LATEST_3_3),str(LATEST_3_4),
-                            str(LATEST_3_5),str(LATEST_3_6),str(LATEST_3_7),str(LATEST_3_8)])
-    def test_compatible_brokers_eos_disabled(self, broker_version):
+    @matrix(broker_version=[str(LATEST_2_8),
```
Comment (chia7712) [inline tests/kafkatest/tests/streams/streams_broker_compatibility_test.py]: 2.8.2 does not support Producer IDs in kraft (see https://github.com/apache/kafka/commit/e97cff2702b6ba836c7925caa36ab18066a7c95d).

File: tests/kafkatest/tests/streams/streams_broker_compatibility_test.py [hunk 3]
```diff
@@ -87,11 +78,12 @@ def test_compatible_brokers_eos_disabled(self, broker_version):
         self.kafka.stop()
 
     @cluster(num_nodes=4)
-    @matrix(broker_version=[str(LATEST_2_5),str(LATEST_2_6),str(LATEST_2_7),str(LATEST_2_8),
+    @matrix(broker_version=[str(LATEST_2_8),
```
Comment (chia7712) [inline tests/kafkatest/tests/streams/streams_broker_compatibility_test.py]: ditto

**Review by chia7712 — COMMENTED** [review]
@bbejeck thanks for this patch. Two major comments remain. PTAL

File: tests/kafkatest/tests/streams/streams_broker_compatibility_test.py (hunk 2 above)
Comment (bbejeck) [reply tests/kafkatest/tests/streams/streams_broker_compatibility_test.py]: ack

**Comment by bbejeck** [comment]
Confirmed 2.8.2 fails in system test per [previous comment](https://github.com/apache/kafka/pull/17603#discussion_r1818118747)

broker_fails_at_2 8

**Review by chia7712 — APPROVED** [review]
LGTM, run `streams_broker_compatibility_test.py` on my local. all pass

**Comment by bbejeck** [comment]
Thanks for merging @chia7712!

[Link to passing system test](https://semaphore.ci.confluent.io/jobs/9bbf9044-bc0e-4d9b-ba5a-eaa243eb48b2/summary?report_id=0356f43b-c807-318c-b61e-09e8fdd4e976)

broker_compatibility_test_results

**Pull request merged**